weblate/trans/models/unit.py

```diff
@@ -2008,10 +2008,16 @@ def update_explanation(
                     unit=unit,
                     author=user,
                 )
-            PendingUnitChange.store_unit_change(
-                unit=self,
-                author=user,
-            )
+
+            # translation file does not exist for source strings in bilingual formats
+            if (
+                not self.is_source
+                or self.translation.component.file_format_cls.monolingual
+            ):
+                PendingUnitChange.store_unit_change(
+                    unit=self,
+                    author=user,
+                )
```

Comment:
Thanks, this is more future-proof than the original logic before a74d58071ea5e632c46de6bd3c370a3606e6cba8 which simply didn't set the pending flag on the source string at all.